lir/aggregation.py

```diff
@@ -30,15 +32,54 @@ def close(self) -> None:  # noqa: B027
         pass
 
 
+class AggregatePlot(Aggregation):
+    """Aggregation that generates plots by repeatedly calling a plotting function."""
+
+    def __init__(self, plot_function: Callable, output_dir: str) -> None:
+        super().__init__()
+
+        self.f = plot_function
+        self.dir = output_dir
+        self.plot_type = plot_function.__name__
+
+        self.plots: dict[str, Any] = {}  # Dictionary to hold figures and axes for different plot types
+        # Add initial plot setup for this plot_type
+        self._setup_plot()
+
+    def _setup_plot(self) -> None:
+        """Set up a new plot for the given plot type."""
+        fig, ax = plt.subplots(figsize=(10, 8))
+        canvas = Canvas(ax)
+        self.plots = {'fig': fig, 'ax': ax, 'canvas': canvas, 'legend_suffix': []}
```

Comment:
Het komt me nog steeds over als omslachtig. Waarom niet gewoon

```suggestion
        self._fig, self._ax = plt.subplots(figsize=(10, 8))
        self._canvas = Canvas(ax)
```